instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/snippet/SnippetInjectingResponseWrapper.java

```diff
@@ -0,0 +1,159 @@
+  private static final int SNIPPET_LENGTH = SNIPPET.length();
+  public static final String FAKE_SNIPPET_HEADER = "FAKE_SNIPPET_HEADER";
+
+  public SnippetInjectingResponseWrapper(HttpServletResponse response) {
+    super(response);
+  }
+
+  @Override
+  public boolean containsHeader(String name) {
+    // override this function in order to make sure the response is wrapped
+    // but not wrapped twice
+    // we didn't use the traditional method req.setattribute
+    // because async may carry out the old req and new resp without wrapped
+    if (name.equals(FAKE_SNIPPET_HEADER)) {
+      return true;
+    }
+    return super.containsHeader(name);
+  }
+
+  @Override
+  public void setHeader(String name, String value) {
+    if (shouldInject() && "Content-Length".equalsIgnoreCase(name)) {
+      try {
+        value = Integer.toString(SNIPPET_LENGTH + Integer.valueOf(value));
+      } catch (NumberFormatException ex) {
+        System.err.println("Invalid string format");
+      }
+    }
+    super.setHeader(name, value);
+  }
+
+  @Override
+  public void addHeader(String name, String value) {
+    if (shouldInject() && "Content-Length".equalsIgnoreCase(name)) {
+      try {
+        value = Integer.toString(SNIPPET_LENGTH + Integer.valueOf(value));
+      } catch (NumberFormatException ex) {
+        System.err.println("Invalid string format");
+      }
+    }
+    super.addHeader(name, value);
+  }
+
+  @Override
+  public void setIntHeader(String name, int value) {
+    if (shouldInject() && "Content-Length".equalsIgnoreCase(name)) {
+      value = SNIPPET_LENGTH + value;
+    }
+    super.setIntHeader(name, value);
+  }
+
+  @Override
+  public void addIntHeader(String name, int value) {
+    if (shouldInject() && "Content-Length".equalsIgnoreCase(name)) {
+      value = SNIPPET_LENGTH + value;
+    }
+    super.addIntHeader(name, value);
+  }
+
+  @Override
+  public void setContentLength(int len) {
+    if (shouldInject()) {
+      len = len + SNIPPET_LENGTH;
+    }
+    super.setContentLength(len);
+  }
+
+  @Nullable private static final MethodHandle setContentLengthLongHandler = getMethodHandle();
+
+  @Nullable
+  private static MethodHandle getMethodHandle() {
+    try {
+      return MethodHandles.lookup()
+          .findSpecial(
+              HttpServletResponseWrapper.class,
+              "setContentLengthLong",
+              MethodType.methodType(void.class),
+              SnippetInjectingResponseWrapper.class);
+    } catch (NoSuchMethodException | IllegalAccessException e) {
+      System.out.println("SnippetInjectingResponseWrapper setContentLengthLong e: " + e);
+      return null;
+    }
+  }
+
+  private boolean shouldInject() {
+    boolean shouldInject = true;
+    if (!super.containsHeader("content-type")) {
+      shouldInject = false;
+    }
+    String contentType = super.getContentType();
+    if (contentType == null || !contentType.contains("text/html")) {
+      shouldInject = false;
+    }
```

Comment:
let's check getContentType() first, then fall back to getHeader("content-type") if null